### PR TITLE
Call network phase2 on retry

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/manager.go [hunk 1]
```diff
@@ -1371,6 +1372,11 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
 	// To make sure, that we set the right qemu wrapper arguments,
 	// we update the domain XML whenever a VirtualMachineInstance was already defined but not running
 	if !newDomain && cli.IsDown(domState) {
+		domain, err = l.preStartHook(vmi, domain)
```
Comment (AlonaKaplan) [inline pkg/virt-launcher/virtwrap/manager.go:1375]: I think you need to invoke just `network.SetupNetworkInterfacesPhase2(vmi, domain)`, not the whole `preStartHook`.

Comment (qinqon) [reply pkg/virt-launcher/virtwrap/manager.go:1375]: But the problem in general is that on retries we don't update the dom xml from libvirt we re-create it again right ? so we have to put back the bits that are set on preStartHook.

**Pull request closed without merging**